### ISSUE TITLE
App Crashing When Setting QBRTCRemoteVideoView

### ISSUE THREAD

**Issue by adil-appifytech** [QuickBlox/quickblox-ios-sdk#752]
**Environment details**
*iOS version, Quickblox iOS SDK version, QuickbloxWebRTC SDK version (optional)*
Quickblox-WebRTC => 2.4
QuickBlox => 2.9.1

**Did this work before?**
YES

**Expected behavior**
I should be able to set QBRTCRemoteVideoView to provided track.

**Actual behavior**
I am notable to set QBRTCRemoteVideoView to provided track.

**Logs**
[group_call_crash.txt](https://github.com/QuickBlox/quickblox-ios-sdk/files/763781/group_call_crash.txt)

<img width="1589" alt="screen shot 2017-02-09 at 6 28 44 pm" src="https://cloud.githubusercontent.com/assets/10629648/22785352/c45c38a2-eef5-11e6-90ad-c7543ec713c6.png">

**Steps to reproduce the behavior**
Make a group call

**Comment by Raikerian** [comment]
Hi,

According to your log, you have already set that video track in another remote video view.

**Comment by adil-appifytech** [comment]
@Raikerian - It's big change for us and we have to figure out how to proceed further.

**Comment by Raikerian** [comment]
Try to debug your code and find where are you using this video track else.

**Comment by adil-appifytech** [comment]
@Raikerian - I know why and where it's happening. Now I want to know how I can better handle this situation. 

In simple words, I want to know how we can reassign a tracker to another renderer and remove from last renderer.

**Comment by adil-appifytech** [comment]
@Raikerian - This issue came when we moved from Webrtc 2.1 to Webrtc 2.3 and further.

**Comment by Raikerian** [comment]
Just call this method of QBRTCVideoTrack and pass your remote view there if you want to delete renderer:

```objc
/** Deregister a renderer. */
- (void)removeRenderer:(id<RTCVideoRenderer>)renderer;
```

**Comment by adil-appifytech** [comment]
@Raikerian - I am doing that but it's not working as we want it to.

**Comment by VitaliyGuru** [comment]
@adil-appifytech,
> This issue came when we moved from Webrtc 2.1 to Webrtc 2.3 and further.

Please specify this in issue Environment details. It's important information.

**Comment by Raikerian** [comment]
Well, thats the only way to remove renderer from video track. Why would you want to change remote view for a single video track anyway?

**Comment by adil-appifytech** [comment]
@Raikerian @VitaliyGuru 

If we have a method to set the tracker like this

```
    /**
     *  Set video track
     *
     *  @param videoTrack QBRTCVideoTrack instance
     */
    open func setVideoTrack(_ videoTrack: QBRTCVideoTrack)
```

Why there is no method to remove tracker here?

**Comment by Raikerian** [comment]
We will add method for remove in next release.
As for removeRenderer seems like it has a bug right now and not really removing anything.
We will fix it in a future release too.

Again, why are you setting different renderers to one videoTrack? Workaround for now would be only to set renderer 1 time. Check our sample in current repository for this.

**Comment by Raikerian** [comment]
WebRTC does not allowing more than 1 renderer for video track for performance reasons, please try to use only 1 remote video view for 1 video track.

**Comment by adil-appifytech** [comment]
Can we have that next release in **next few hours?**

Below is our issue

![img_0040](https://cloud.githubusercontent.com/assets/10629648/22826686/6d9663c4-efb5-11e6-9a46-436942331f2b.jpg)

**Comment by Raikerian** [comment]
Just don't set video track in that remote video view, but set remote video view with track to that view. That is the fix to your issue. That small view should not be remote video view itself, but view that will eventually show appropriate remote video view within it.

Here is the example of how we are doing it in sample:
https://github.com/QuickBlox/quickblox-ios-sdk/blob/master/sample-videochat-webrtc/sample-videochat-webrtc/ViewControllers/CallViewController/CallViewController.m#L443-L454

**Comment by adil-appifytech** [comment]
I have done what you have said but video is not showing at all in that case.

**Comment by adil-appifytech** [comment]
@Raikerian - Is it possible to change the frame of QBRTCRemoteVideoView later?

When I can have the new sdk with fixes?

**Comment by Raikerian** [comment]
Fixed SDK will be attached to your issue.
Yes you can change frame.

**Comment by Raikerian** [comment]
SDK with fixed removeRenderer method in video rack class was added to your ticket and should be delivered to you by support team now.

**Comment by adil-appifytech** [comment]
@Raikerian - Thanks.

**Comment by Pro100Andrey** [comment]
@adil-appifytech, fixed v2.4.1
